runtime/router/router.go

```diff
@@ -95,7 +96,22 @@ func (r *Router) Handle(method, path string, handler http.Handler) error {
 		trie = NewTrie()
 		r.tries[method] = trie
 	}
-	return trie.Set(path, handler, r.isWhitelistedPath(path))
+	err := trie.Set(path, handler, r.isWhitelistedPath(path))
+	if err == errExist {
+		return &urlFailure{url: path, method: method}
+	}
+	return err
+}
+
+// urlFailure captures errors for conflicting paths
+type urlFailure struct {
+	method string
+	url    string
+}
+
+// Error returns the error string
+func (e *urlFailure) Error() string {
+	return fmt.Sprintf("panic: path: %q method: %q conflicts with an existing path", e.url, e.method)
```

Comment:
don't think `panic` should be part of the message itself.

Reply:
acked. we can improve message in future diff.